infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/DefaultClusterManager.java

```diff
@@ -189,6 +190,14 @@ public void shutdown() {
         log.info("Shutdown Cluster Manager completed.");
     }
 
+    @Override
+    public String getLocalNodeId() {
```

Comment:
can we add a getter on localNodeId and remove this method?

Reply:
ok.

This seems a bit roundabout.. DiscoveryService -> ClusterManager -> DiscoveryService to fetch the localEndpoint.  Any reason for changing it?

Reply:
agreed, its a round about here...but only here though, ITs only
There is no other way I could think of getting the localNodeId without having the endpoint.

Applications don't need the endpoint to compute the nodeId.